### DISCUSSION TITLE
Hasura Cloud - Custom Supabase Schemas Not Showing Up

### DISCUSSION THREAD

**Discussion by oagix** [hasura/graphql-engine#8487] [Question] [answered]
Hey,

We're using Hasura with Supabase to consume and secure the data. When using our custom schema and its tables, the schema does not appear in the Hasura Cloud interface.

When trying to create it though, it says "Schema already exists", so it is able to read it from Postgres.

Any clue if there's an extra step we need to take to have the schema be accessible via the API using the "service_role" properly? It only shows the default schemas, but none of the custom ones, even if all permissions were set, etc.

Is this a Hasura Cloud UI bug?

**Comment by seanonthenet** [comment]
I'm not familiar with Supabase at all, but from their site it looks like it should support standard Postgres databases. 
Do you have an existing database connection in that Hasura instance?
Are you able to access these custom schemas with another SQL client?

**Comment by akhil-naidu** [comment, marked as answer]
If you are connecting to a Postgres dB with some data or tables, you need to step ahead and tell Hasura which tables you would like to track using Hasura.

In the case of Subpabase, you will have some prepopulated tables, so this is what you have to follow, these steps.

#### All of your traditional tables, other than features provided by the subpabase will be stored in the `Public Folder`
<img width="1512" alt="Screenshot 2022-06-15 at 3 12 37 PM" src="https://user-images.githubusercontent.com/11688537/173800129-fe0fe7bd-8eaa-470a-bcf1-af458d3d98ee.png">

#### Click on `track all` or track only specific tables for which you would like to create GraphQL APIs, which are also termed as Hasura Views
<img width="1512" alt="Screenshot 2022-06-15 at 3 20 39 PM" src="https://user-images.githubusercontent.com/11688537/173800315-53b466d2-8c65-4bdf-90a9-9588694997c8.png">

#### A quick comparison of Hasura API explorer(Which allows you see all the Hasura views) before and after tracking
> Before Tracking the table `test`
<img width="1512" alt="Screenshot 2022-06-15 at 3 20 09 PM" src="https://user-images.githubusercontent.com/11688537/173800371-0f7cfc08-78bc-403e-8f1f-376ecfe16f25.png">

> After Tracking the table `test`
<img width="1512" alt="Screenshot 2022-06-15 at 3 21 35 PM" src="https://user-images.githubusercontent.com/11688537/173800411-fd9d32d2-b025-4b56-beae-da100be59c39.png">

---

**Note**: Hasura allows you to create and configure your tables from the Hasura console itself, but for some reason, if you prefer using Supabase to create and configure your table schemas, it is mandatory to track them back in the Hasura console.

The reason behind this architectural decision is practical for the majority of the Postgres developers; not every Postgres developer expects to have a CRUD operation on every existing Postgres table. Tracking tables individually would give them more control on whether to expose a specific table to the Pubic GraphQL API or not.
